services/adminService.ts

```diff
@@ -0,0 +1,212 @@
+import { v4 as uuidv4 } from 'uuid';
+import { Response, Request } from 'express';
+import { IGetUserAuthInfoRequest, Books, User, Users } from "../serviceTypes"
+import { RES_STATUS } from '../constants';
+import { adminQueries } from '../database/dbQueries';
+import { databaseQuery } from '../database/db_queryFunction';
+import { QueryResult } from 'pg'
+
+
+/**
+ * This function is to add books in db by librarian i.e admin
+ * and returns Books object.
+ * @param {Request["body"]} body is req.body which is an object or undefined.
+ * @param {User} reqUser is req.user which is an object of type User. 
+ * @returns {Promise<Books>} Books is an object.   
+*/
+export const addBook = async (body: Request["body"], reqUser: User): Promise<Books> => {
```

Comment:
nit: can we make this type to `Request.body`

Reply:
No because it will give error it cannot be used for type.